### DISCUSSION TITLE
Is the latest version work with NextJS latest now?

### DISCUSSION THREAD

**Discussion by huypham1411** [ant-design/ant-design#43505] [Q&A] [answered]
I just wondering if the latest build, Antd v5.7.0 work well with Next v13.4.9? I've just upgraded both to the latest version and found out that the font family is gone. I'm not sure if this issue occurs because of Antd or Next so I tried rolling back the previous version and everything back to normal. Hope someone can help

> My current package.json
"dependencies": {
    "@ant-design/icons": "^5.1.0",
    "@hookform/resolvers": "^3.1.0",
    "@reduxjs/toolkit": "^1.9.5",
    "@types/node": "20.4.1",
    "@types/react": "^18.2.14",
    "@types/react-dom": "^18.2.6",
    "add": "^2.0.6",
    "antd": "^5.6.4",
    "axios": "^1.4.0",
    "classnames": "^2.3.2",
    "dayjs": "^1.11.9",
    "env-cmd": "^10.1.0",
    "eslint": "8.44.0",
    "eslint-config-next": "13.4.9",
    "i18next": "^23.2.10",
    "lodash": "^4.17.21",
    "moment": "^2.29.4",
    "next": "13.4.7",
    "querystring": "^0.2.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-hook-form": "^7.43.9",
    "react-hotkeys-hook": "^4.4.1",
    "react-number-format": "^5.2.2",
    "react-redux": "^8.0.5",
    "sass": "^1.62.1",
    "sharp": "^0.32.2",
    "swr": "^2.1.5",
    "typescript": "5.1.6",
    "yarn": "^1.22.19",
    "yup": "^1.1.1"
  },
  "devDependencies": {
    "@types/lodash": "^4.14.195",
    "autoprefixer": "^10.4.14",
    "postcss": "^8.4.25",
    "tailwindcss": "^3.3.2"
  }

**Comment by zombieJ** [comment]
Hi, 
Since NextJS has some hook for popular front end framework compile optimization, it may not align with the latest version. But we are all working on it. Pls feel free to provide reproduce : )

**Reply by huypham1411** [reply]
As I did some experiments, Antd v5.6.4 will have this error: 

> Module not found: Can't resolve 'antd/es/ConfigProvider'
 
with Next v13.4.8 and seem like the latest Antd version solves this error as  #this https://github.com/ant-design/ant-design/issues/43362 so I tried to test the latest AntD version with Next v13.4.8 and the error still occurs so I tried it with Next v13.4.9. Start a project without any errors but as I mentioned above, the font-family seems to be lost. My current project uses Roboto but the font-family after upgrading both the library and the framework to the latest version looks like the font-family now is Times New Roman instead. I'll try my best to provide the reproduce step

**Reply by vitalijalbu** [reply]
show me you app.js code

**Reply by huypham1411** [reply]
here is my page.tsx code from app directory

```
'use client'

import { path } from '@/constants'
import { getAccessTokenFromLS } from '@/utils'
import { redirect } from 'next/navigation'
import { Fragment, useEffect } from 'react'

export default function Home() {
  const accessToken = getAccessTokenFromLS()

  useEffect(() => {
    if (!accessToken) {
      redirect(path.login)
    } else {
      redirect(path.findPart)
    }
  }, [accessToken])

  return <Fragment></Fragment>
}
```

**Reply by huypham1411** [reply]
and here is the layout.tsx
```
import '@/styles/styles.scss'
import { Roboto } from 'next/font/google'
import { AppProvider } from '@/contexts'

const roboto = Roboto({ weight: ['300', '400', '500', '700'], subsets: ['latin'] })

export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <html suppressHydrationWarning={true}>
      <body className={roboto.className} suppressHydrationWarning={true}>
        <AppProvider>{children}</AppProvider>
      </body>
    </html>
  )
}

```

**Reply by vitalijalbu** [reply]
i need also appprovider

**Reply by huypham1411** [reply]
here is the AppProvider
```
'use client'

import { App } from 'antd'
import { AntdProvider } from './antd-provider'
import { ReduxProvider } from './redux-provider'
import { SWRProvider } from './swr-provider'

export function AppProvider({ children }: { children: React.ReactNode }) {
  return (
    <App>
      <ReduxProvider>
        <SWRProvider>
          <AntdProvider>{children}</AntdProvider>
        </SWRProvider>
      </ReduxProvider>
    </App>
  )
}
```

**Reply by vitalijalbu** [reply]
AntdProvider. also ... bro why having all these layouts separated, when you can he only 2 well structured??

**Reply by huypham1411** [reply]
here is the antd provider
```
'use client'

import { ConfigProvider } from 'antd'
import { theme } from '@/utils'

export function AntdProvider({ children }: { children: React.ReactNode }) {
  return <ConfigProvider theme={theme}>{children}</ConfigProvider>
}
```
it's my first time setup NextJS project so it's a bit complicated sorry

**Reply by vitalijalbu** [reply]
so basically the issue is the font? why you dont add the font family inside theme.json of antd?

**Reply by huypham1411** [reply]
I think my teamate had already done this as I found this theme.ts file
```
export const theme = {
  token: {
    colorPrimary: '#3651d9',
    colorError: '#EB5146',
    fontFamily: '__Roboto_5506ec, __Roboto_Fallback_5506ec'
  }
}
```
and this was imported to the antd provider, not sure if this is what you intended to tell me

**Reply by vitalijalbu** [reply]
need json externsion, not ts see here the theme editor, after export it: https://ant.design/theme-editor

**Reply by huypham1411** [reply]
So I change the theme file to json and update its contain as below
```
{
  "token": {
    "colorPrimary": "#3651d9",
    "colorError": "#EB5146",
    "fontFamily": "__Roboto_5506ec, __Roboto_Fallback_5506ec"
  }
}
```
and I've updated the antd provider as follow
```
'use client'

import { ConfigProvider } from 'antd'
import * as theme from '@/utils/theme.json'

export function AntdProvider({ children }: { children: React.ReactNode }) {
  return <ConfigProvider theme={theme}>{children}</ConfigProvider>
}
```
but the font is still Times New Roman instead of Roboto

**Reply by huypham1411** [reply]
hey, here is the quick reproduce https://codesandbox.io/p/sandbox/beautiful-platform-7ztdyk?file=%2Fapp%2Flayout.tsx%3A16%2C12 hope you can take a look, thanks

**Reply by zombieJ** [reply]
https://github.com/ant-design/ant-design/issues/43362 `ConfigProvider` path error is a bug of `Next.JS@13.4.8`. If meet this, run `npm ls next` to check the version which installed.

**Comment by huypham1411** [comment]
> Hi, Since NextJS has some hook for popular front end framework compile optimization, it may not align with the latest version. But we are all working on it. Pls feel free to provide reproduce : )

@zombieJ 
here is the quick reproduce 
https://codesandbox.io/p/sandbox/beautiful-platform-7ztdyk?file=%2Fapp%2Flayout.tsx%3A16%2C12
hope you can take a look, thanks!

**Reply by huypham1411** [reply]
oh and I just want to mention that the font-family works normally with AntD v5.6.5 and NextJS v13.4.7, so I don't think the theme file extension is the problem here as I had tried to update the file and still got the font-family issue

**Reply by zombieJ** [reply]
What's the font family gone means. Seems Input has correct style:


<img width="557" alt="截屏2023-07-13 11 24 18" src="https://github.com/ant-design/ant-design/assets/5378891/30c3981b-e3c9-46e5-91ec-2e2c64fb5805">

<img width="401" alt="截屏2023-07-13 11 47 45" src="https://github.com/ant-design/ant-design/assets/5378891/b5ed4cf1-9cf9-4b55-84ba-9bde10c5beec">

**Reply by huypham1411** [reply]
I am expecting the font-family to be Roboto as I had config it in the theme.json file but got Times New Roman instead. As I mentioned above, the font-family still work normally (Roboto font-family as config) with AntD v5.6.5 and NextJS v13.4.7, after upgrading both to the latest, the font-family changed so I'm trying to figure out the reason behind this

**Reply by zombieJ** [reply]
If you mean `username:` font not correct. It's from the order not correct:

```tsx
<App>
  <AntdProvider>
    {children}
  </AntdProvider>
</App>
```

which should be:
```tsx
<AntdProvider>
  <App>
    {children}
  </App>
</AntdProvider>
```

<img width="400" alt="截屏2023-07-13 11 51 38" src="https://github.com/ant-design/ant-design/assets/5378891/a2a2047b-d8f1-4d52-b233-b0a9153ec6cc">

**Reply by zombieJ** [reply]
In CSSINJS, styles have the same priority as css selectors. So if you encounter problems caused by version changes, it is likely to be caused by changes in the order of dynamic insertion styles. In your example, Roboto's root node priority and antd's App style priority are also the same. So I speculate that in order not to overwrite the used style, next inserts its own style to the front so that App can overwrite it.

**Reply by huypham1411** [reply]
Here is what my site look like before upgrading
![Screen Shot 2023-07-13 at 10 53 25 AM](https://github.com/ant-design/ant-design/assets/54835897/196a2aa4-9513-47b3-91ec-321bf92a765a)
and here is after the upgrade
![Screen Shot 2023-07-13 at 10 53 48 AM](https://github.com/ant-design/ant-design/assets/54835897/0a5ee370-36eb-4b2f-9617-73e6e0bc2587)

**Reply by zombieJ** [reply]
> If you mean username: font not correct. It's from the order not correct:

Could you help check this? I guess it may help

**Reply by zombieJ** [reply]
You can inspect the element, find out which style override the fontFamily. It may help for debug.

**Reply by huypham1411** [reply]
here is the before-upgrading element inspect
![Screen Shot 2023-07-13 at 11 17 13 AM](https://github.com/ant-design/ant-design/assets/54835897/7ab3b439-2a25-4f32-a8b4-7b68815d7f64)
here is the after inspect
![image](https://github.com/ant-design/ant-design/assets/54835897/b57fb647-7216-4cbf-b7c5-8d94c85a4992)

**Reply by zombieJ** [reply]
The fontFamily does be the `Roboto_xxx`. I guess is the `Roboto_xxx` font not loaded (maybe network break or something)

**Reply by huypham1411** [reply]
the only difference between those 2 is after upgrading the css font got this class `css-dev-only-do-not-override` and while all the APIs still working, I don't think it's the network issue

**Reply by zombieJ** [reply]
`css-dev-only` exist in all v5 of antd. That's not the key. It set the `fontFamily` as your screenshot provided. You can update the reproduce with the break demo which I can help on dig this : )

**Reply by huypham1411** [reply]
hey, I've just changed the wrap order of the provider (Bring Antd Provider outside) and seems like the font-family did update but somehow it's kinda weird so I try to check the font using Find font extension for chrome to inspect the text and got this result 
![image](https://github.com/ant-design/ant-design/assets/54835897/b2f2ea2a-99a7-4028-a5bc-18d0ac84bf77)
so I tried another experiment with `NextJS v13.4.7` and `AntD v5.6.4` and well the result is unexpected 
![image](https://github.com/ant-design/ant-design/assets/54835897/0f2ba591-0a3e-4bf8-b22a-34d4e3fe604c)
so can you help me check step by step like this 
test the font-family with both AntD and NextJS in the latest version
test the font-family again with `NextJS v13.4.7` and `AntD v5.6.4`
you can still check the codesandbox link I provided
just edit the package.json with the version then run yarn to install the package then yarn dev to run
both tests above on the same input component

**Reply by huypham1411** [reply]
@zombieJ  can you help me check? Thanks

**Reply by zombieJ** [reply, marked as answer]
https://github.com/ant-design/ant-design/issues/43547#issuecomment-1635220518

**Reply by huypham1411** [reply]
but I wonder why upgrading the version make this change

**Reply by huypham1411** [reply]
and how can I keep the previous font-family when upgrading the framework and library to the latest version?
